smart_card_connector_app/src/mock-chrome-api.js

```diff
@@ -238,5 +238,20 @@ GSC.ConnectorApp.MockChromeApi = class {
         `Event to result function map does not have ${eventName}`);
     return chrome.smartCardProviderPrivate[EVENT_TO_RESULT_MAP[eventName]];
   }
+
+  /**
+   * Switches the mocked function back to record mode.
+   * Verifies that there were no unmet expectations before switching.
+   * @param {string} functionName
+   * @returns {Function} Mocked function object.
+   */
+  record(functionName) {
```

Comment:
I was really confused by this name for a bit, because my expectation on reading the name was that this would record an actual function call. Instead, it is more of a reset with extras. I see that you chose record as a name in contrast to replay - but it does behave differently from replay notably in that it returns a mock, while replay is called on a mock and returns void.

What this does is a verify/reset/get. This feels more like a getNewMock() method to me, or maybe getFresh() or getFreshMock(), or maybe just mock() - WDYT?

Also, in your commit message you claim that the testing framework doesn't offer a way to switch between replay and record mode - but isn't this verify/reset exactly that?

Reply:
>What this does is a verify/reset/get. This feels more like a getNewMock() method to me, or maybe getFresh() or getFreshMock(), or maybe just mock() - WDYT?

Thanks for the suggestion, updated the name to `getFreshMock`.

>Also, in your commit message you claim that the testing framework doesn't offer a way to switch between replay and record mode - but isn't this verify/reset exactly that?

Yep, but it's more of a hack probably.